docs/cookbook/creating-custom-schema-fields.md

```diff
@@ -0,0 +1,999 @@
+
+This method takes four arguments from the Vue component. The two most important for this recipe are the `data` and `object` arguments. `data` is going to contain all of the data being delivered from the input fields. We are only interested in managing the data from our new custom field, so in the very first line of the method we reassign `data` as being equal to only the data coming from our field using `data = data[field.name]`.
+
+We next pull in the schema we defined above using the `exposeSchema()` method. This schema is used to inform the `convert()` method of what types of data laundering methods to use.
+
+The `errors` and `results` variables are next initialized so that they can take values returned from the `@apostrophecms/schema` module `convert()` method.
+
+Note that for `result` we are making sure that an `_id` property exists and is sanitized, otherwise, we generate one.
+
+Before passing our data to the `convert()` method we ensure that it is a non-array object. If not, the data coming back from the component can't be sanitized and instead is discarded.
+
+
+```javascript
+async convertInput(req, field, data, object) {
+  ...
+}
+try {
+  await self.apos.schema.convert(req, schema, data, result);
+} catch (e) {
+  for (const error of e) {
+    errors.push({
+      path: error.path,
+      error: error.error
+    });
+  }
+}
+object[field.name] = result;
+if (errors.length) {
+  throw errors;
+}
+```
+
+After preping the argument data, we invoke the `@apostrophecms/schema` module `convert()` method within a try-catch block.
+
+This either populates our `result` variable with the sanitized data, or returns errors that are pushed to the `errors` variable to be thrown at the end.
+
+Lastly, we assign those results to the `object` parameter. If no errors came back from the `convert()` method this object is saved to the database.
+
+```javascript
+extendMethods(self) {
+  return {
+    getBrowserData(_super, req) {
+      const browserData = _super(req);
+      browserData.gradeSchema = self.exposeSchema();
+      return browserData;
+    }
+  };
+}
+```
+
+The last module configuration function that we are setting for this module is `extendMethods(self)`. In order to expose our schema to the Vue component, we need to add it to the `req` HTTP request object. This is accomplished through the `getBrowserData()` method. In this case, we don't want to replace the base `@apostrophecms/piece-type` method, only extend it.
+
+The `_super` argument is passing the original function (you can read more about this pattern in the [documentation](/reference/module-api/module-overview.html#the-extension-pattern)). The original method takes a single argument, `req`, that is passed as the second argument.
+
+The `browserData` variable is set to the original value of `req` by invoking the original method with the `req` argument. That object is then updated with the custom schema array and the new `req` returned, just like the original method.
```

Comment:
It is set to the return value of the original method. Not the original value of `req`.

Reply:
We're not returning `req` here. We're returning data specific to this module to be made available when this request ends by rendering a page.

Reply:
CHanged.